src/mbed_tools/cli/list_connected_devices.py

```diff
@@ -49,25 +49,40 @@ def _sort_devices(devices: Iterable[Device]) -> Iterable[Device]:
     return sorted(devices, key=attrgetter("mbed_board.board_name", "serial_number"))
 
 
+def _get_ids_devices(devices: Iterable[Device]) -> Iterable[Tuple[Optional[int], Device]]:
```

Comment:
We should be more specific about the return type, if it's a list we should specify that. An `Iterable` could be list, generator, string, etc.